src/repr/src/scalar.rs

```diff
@@ -575,10 +575,16 @@ impl<'a> Datum<'a> {
                     (Datum::Array(array), ScalarType::Array(t)) => {
                         array.elements.iter().all(|e| match e {
                             Datum::Null => true,
-                            Datum::Array(_) => is_instance_of_scalar(e, scalar_type),
                             _ => is_instance_of_scalar(e, t),
                         })
                     }
+                    (Datum::Array(array), ScalarType::Int2Vector) => {
+                        array.dims().len() == 1
```

Comment:
Does this mean that `int2vector` doesn't support multi-dimensional arrays? If so, we may want to add a test to show that it gracefully returns an error in those cases.

Reply:
Ah, there's a complexity here in that multi-dimensional arrays cannot be input as `int2vector`, so there's no way to test them. This is meant only to catch a case where we might accidentally somehow let a 2D array into a position where we expect an `int2vector`. What you're pointing to, though, is the inherent vice in having one datum support multiple scalar types, and a good point, but something that we're generally willing to accept.